cmd/controller/main.go

```diff
@@ -43,6 +54,11 @@ func main() {
 	lo.Must0(op.AddHealthzCheck("cloud-provider", awsCloudProvider.LivenessProbe))
 	cloudProvider := metrics.Decorate(awsCloudProvider)
 
+	err := validateK8sVersion(ctx, op)
```

Comment:
nit: If you can collapse this down to one line (i.e. you don't need the err somewhere else), it's generally nice to do so

```go
if err := validateK8sVersion(ctx, op); err != nil {
  logging.FromContext(ctx).Error(err)
}
```